server/datastore/mysql/apple_mdm.go

```diff
@@ -6317,6 +6332,11 @@ VALUES (?, ?, ?, ?, ?, ?, ?, ?)
 		return nil, ctxerr.Wrap(ctx, err, "encrypt abm_token with datastore.serverPrivateKey")
 	}
 
+	// Check that RenewAt is not zero
+	if tok.RenewAt.IsZero() {
```

Comment:
nit: Don't we want to do this check earlier?, so that we avoid doing the unnecessary `encrypt` call on [this line](https://github.com/fleetdm/fleet/pull/31445/files#diff-2f8a6d69f21403d6845da98a6a9a7e3808f275a21ec0209cbabcdbeaae57676eR6330)